OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -266,43 +266,30 @@ bool Joint::isCoordinateUsed(const Coordinate& aCoordinate) const
 
 void Joint::scale(const ScaleSet& scaleSet)
 {
-    SimTK::Vec3 parentFactors(1.0);
-    SimTK::Vec3 childFactors(1.0);
-
-    // Find the factors associated with the PhysicalFrames this Joint connects
-    const string& parentName = getParentFrame().findBaseFrame().getName();
-    const string& childName = getChildFrame().findBaseFrame().getName();
-    // Get scale factors
-    bool found_p = false;
-    bool found_b = false;
-    for (int i = 0; i < scaleSet.getSize(); i++) {
-        Scale& scale = scaleSet.get(i);
-        if (!found_p && (scale.getSegmentName() == parentName)) {
-            scale.getScaleFactors(parentFactors);
-            found_p = true;
-        }
-        if (!found_b && (scale.getSegmentName() == childName)) {
-            scale.getScaleFactors(childFactors);
-            found_b = true;
+    auto scaleOffsetFrameIfOwned = [&](const PhysicalFrame& frame) -> void {
+        // If the frame is owned by this Joint then scale it.
+        // Otherwise let the owner of the frame decide.
+        if (auto* offset = findComponent<PhysicalOffsetFrame>(frame.getName()))
```

Comment:
It doesn't seem like this line does what the comment above advertises. Doesn't there need to be a check to satisfy the condition "If the frame is owned by this Joint"?

Reply:
This code will ensure that the frame is an "ownership" descendant of the Joint, but does not ensure that the frame is a direct descendant (immediate subcomponent). Which of these two meanings is meant by "frame is owned by this Joint"?

Reply:
I don't think it really matters where the frame lives inside the `Joint` as long as it is the *child* or *parent* frame. If it is some other frame it should not get scaled in this scheme.